### PR TITLE
Add shaders without user-defined parameters.

### LINKED ISSUE #12: Create custom shader transformation node
As a VC user, I want to create a custom transformation node with my shader, to transform frames accordingly to my specific business requirements.

### PR DESCRIPTION
**Pull request by jerzywilczek**
This patch allows shaders that don't take custom parameters.

Closes #12, finally

### CODE REVIEW

File: compositor_render/src/transformations/shader/node.rs [hunk 1]
```diff
@@ -14,18 +14,27 @@ pub struct ShaderNode {
...
+                .device
                 .create_buffer_init(&wgpu::util::BufferInitDescriptor {
-                    label: Some("shader node custom params buffer"),
+                    label: Some("shader node empty custom params buffer"),
+                    contents: &[0],
                     usage: wgpu::BufferUsages::UNIFORM,
-                    contents: &params,
-                });
+                }),
+        };
```
Comment (WojciechBarczynski) [inline compositor_render/src/transformations/shader/node.rs:37]: With this, uniform in shader still has to be defined, right? I don't think defining `custom_params` (or sth like this) uniform in the shader when no parameters are used is a good solution. Maybe we should have two different bind groups layouts etc. for that.

**Review by wkozyra95 — APPROVED** [review]

File: compositor_render/src/transformations/shader/node.rs (hunk 1 above)
Comment (jerzywilczek) [reply compositor_render/src/transformations/shader/node.rs:37]: i was a bit surprised by this, but the shader works without defining this uniform

Comment (jerzywilczek) [reply compositor_render/src/transformations/shader/node.rs:37]: also, [the spec](https://www.w3.org/TR/webgpu/#abstract-opdef-validating-gpuprogrammablestage) seems to confirm that shaders are allowed to not define bindings for some of the provided bind group entries.

This explains how bindings defined in a shader are validated:
> - For each binding that is [statically used](https://www.w3.org/TR/webgpu/#statically-used) by descriptor:
>    - [validating shader binding](https://www.w3.org/TR/webgpu/#abstract-opdef-validating-shader-binding)(binding, layout) must return true.

But the bindings that are not used by the shader are not validated, so they can be present and actually set to anything really

**Review by WojciechBarczynski — APPROVED** [review]

File: compositor_render/src/transformations/shader/node.rs (hunk 1 above)
Comment (WojciechBarczynski) [reply compositor_render/src/transformations/shader/node.rs:37]: Ok, I didn't know/noticed that :)

File: compositor_render/examples/silly.wgsl [hunk 2]
```diff
@@ -27,7 +27,6 @@ struct CompositorStruct {
 }
 
 @group(0) @binding(0) var textures: binding_array<texture_2d<f32>, 16>;
-@group(1) @binding(0) var<uniform> shaders_custom_buffer: CustomStruct;
```
[the remark this answers is not in the document]
Comment (jerzywilczek) [reply compositor_render/examples/silly.wgsl:30]: oh, true

**Review by noituri — APPROVED** [review]

**Pull request merged**